### PR TITLE
refactor(NodeAuthoring): Extract EditNodeTitleComponent

### PR DESCRIPTION
**Pull request by hirokiterashima**
## Changes
- Componentize EditNodeTitleComponent (used to be a part of NodeAuthoringComponent) 
- Change input label for groups from "Activity Title" to "Lesson Title"
- Add tests

## Test
- Lesson title and step title can be edited and saved as before
   - Lesson title input label should say "Lesson title [position]:"
   - Step title input label should say "Step title [position]:"

### CODE REVIEW

File: src/assets/wise5/authoringTool/node/edit-node-title/edit-node-title.component.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<mat-form-field class="step-title form-field-no-hint">
+  <mat-label *ngIf="!isGroupNode" i18n>Step Title {{ nodePosition }}:</mat-label>
+  <mat-label *ngIf="isGroupNode" i18n>Lesson Title {{ nodePosition }}:</mat-label>
+  <input
+    matInput
+    [(ngModel)]="nodeJson.title"
+    [ngModelOptions]="{ debounce: 500 }"
```
Comment (geoffreykwan) [inline src/assets/wise5/authoringTool/node/edit-node-title/edit-node-title.component.html]: The debounce doesn't seem to work. Maybe try performing the debounce like we do in [label-authoring.component.ts](https://github.com/WISE-Community/WISE-Client/blob/92c3c16e0fdd0c5dffaf6253c1d366440be0f63a/src/assets/wise5/components/label/label-authoring/label-authoring.component.ts#L36-L40).

File: src/assets/wise5/authoringTool/node/edit-node-title/edit-node-title.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    component = fixture.componentInstance;
+    const node = new Node();
+    node.title = 'First step';
+    component.node = node;
+    spyOn(TestBed.inject(TeacherProjectService), 'getNodeById').and.returnValue(node);
+    spyOn(TestBed.inject(TeacherProjectService), 'getNodePositionById').and.returnValue('1.1');
+    component.ngOnChanges();
+    fixture.detectChanges();
+  });
+  it('should show step position and title', () => {
```
Comment (geoffreykwan) [inline src/assets/wise5/authoringTool/node/edit-node-title/edit-node-title.component.spec.ts]: Change 'should show' to 'shows'.

**Review by geoffreykwan — CHANGES_REQUESTED** [review]
Functionality mostly works.

**Comment by hirokiterashima** [comment]
The input wasn't debouncing in the develop branch. Since this is a refactoring PR, I removed the debounce ngModelOption instead of trying to get it to work again. Let's discuss as a group if we want to add debounce back for this input. If we do, we can do that in a future task. 

PTAL

**Review by geoffreykwan — APPROVED** [review]
Looks good.

**Pull request merged**